types/ember-data/index.d.ts

```diff
@@ -265,56 +265,70 @@ export namespace DS {
      * `errors` property on the record.
      */
     class InvalidError extends AdapterError {
-        constructor(errors: any[]);
+        constructor(errors: any[], message?: string)
     }
     /**
      * A `DS.TimeoutError` is used by an adapter to signal that a request
      * to the external API has timed out. I.e. no response was received from
      * the external API within an allowed time period.
      */
-    class TimeoutError extends AdapterError {}
+    class TimeoutError extends AdapterError {
+        constructor(errors: any[], message?: string)
+    }
```

Comment:
This `constructor` declaration (and all the others like it) should be unnecessary, as it’s just inherited directly from the `AdapterError`. See [this playground][pg] to confirm.

[pg]: https://www.typescriptlang.org/play/#code/CYUwxgNghgTiAEkoGdnwILCgBwC4hgFEYYB7GeAbwFgAoeBxUgO2VxgFcxdyAKAsjGQAueFGYBPANoBdADTwAtiFRQA5iAD8otjACWzNQEoA3HQC+dOqCRxE0VPAAqe5aQ65ig+CAAe+ZmA0TBx8IhJyKnhLWjpmEAB3Z1cQd08ImF5ZIzjE5LcPLz5ZBQByeKgeUqMgA

Reply:
@chriskrycho yes that is initially what I have done, unfortunately I ran into an issue with doing `DS.AdapterError.extend`. it gets a compiler error below because `AdapterError` currently extends `Ember.Object`:
```
ERROR: 8:26  expect  TypeScript@4.0 compile error: 
The 'this' context of type 'typeof AdapterError' is not assignable to method's 'this' of type 'typeof AdapterError & (new (properties?: object | undefined) => AdapterError) & (new (...args: any[]) => AdapterError)'.
  Type 'typeof AdapterError' is not assignable to type 'new (properties?: object | undefined) => AdapterError'.
    Types of parameters 'errors' and 'properties' are incompatible.
      Type 'object | undefined' is not assignable to type 'any[]'.
        Type 'undefined' is not assignable to type 'any[]'.
```